core/trino-main/src/main/java/io/trino/operator/BigintGroupByHash.java

```diff
@@ -387,22 +385,19 @@ public AddPageWork(Block block)
         public boolean process()
         {
             int positionCount = block.getPositionCount();
-            checkState(lastPosition < positionCount, "position count out of bound");
```

Comment:
If the page is full RLE then it will not land here. At least one block must be non-RLE. Is there a possibility of that kind of page to be that huge, e.g >10k positions? I assume that page is by-definition a chunk of data that can be processed atomically.

Reply:
>  Is there a possibility of that kind of page to be that huge, e.g >10k positions?

You are betting that every place in code will adhere to these rules and not produce extremely large pages. Additionally, we don't have a control over all connectors, which can potentially produce large pages too. It's better to be defensive here

Reply:
I added batching. No performance difference